src/dev_server.rs

```diff
@@ -4,31 +4,73 @@ use clap::Parser;
 use std::{
     fs,
     io::{prelude::*, BufReader},
-    net::{IpAddr, SocketAddr, TcpListener, TcpStream},
+    net::{IpAddr, Ipv4Addr, SocketAddr, TcpListener, TcpStream},
     path::Path,
     process,
 };
 
+/// A simple HTTP server that can, optionally, watch a given command.
+///
+/// Get the files at `watch_path` and serve them at a given IP address
+/// (127.0.0.1:8000 by default). An optional command can be run along the
+/// server, relaunched if changes are detected in your project via a [`Watch`].
+///
+/// # Usage
+///
+/// ```no_run
+/// use xtask_wasm::{metadata, DevServer};
+/// use std::process::Command;
+///
+/// let build_path = metadata().workspace_root.join("build");
+/// let mut command = Command::new("cargo");
```

Comment:
This is generally not how people will use this crate. You need to make things apparent that people will need to match their cli to retrieve the DevServer object.

Use `#` at the beginning of the code to hide code that is there only for the purpose of making the test pass but is be irrelevant to the user.

Take this for example in the documentation of `tracing`:
https://github.com/tokio-rs/tracing/blob/9d8d366/tracing/src/lib.rs#L180-L193

Only the `use tracing` and the `let json = ...` lines will be visible as you can see here in their doc: https://docs.rs/tracing/0.1.29/tracing/#spans-1